Content.Server/PDA/Ringer/RingerSystem.cs

```diff
@@ -0,0 +1,182 @@
+            {
+                _pathIndex.Add(((Note)indexCount).ToString(), string.Concat("/Audio/Effects/RingtoneNotes/", ((Note)indexCount).ToString().Replace("sharp", "#").ToLower()) + ".ogg");
+            }
+        }
+
+        //Event Functions
+
+        private void RingerPlayRingtone (EntityUid uid, RingerComponent ringer, RingerPlayRingtoneMessage args)
+        {
+            ringer.IsPlaying = true; //Allows the overridden Update function to play the ringtone based off of the ringer component's stored ringtonePathArr
+            UpdateRingerUserInterface(ringer);
+        }
+
+        private void UpdateRingerUserInterfaceDriver(EntityUid uid, RingerComponent ringer, RingerRequestUpdateInterfaceMessage args)
+        { //Seprate Drivers are needed because UpdateRingerUserInterface() and UpdateRingerComponent() are called by other functions aside from the events themselves
+            UpdateRingerUserInterface(ringer);
+        }
+
+        private void RingerUpdateRingtoneDriver(EntityUid uid, RingerComponent ringer, RingerSetRingtoneMessage args)
+        {
+            Note[] _RingtoneNoteArgumentArray = new Note[4];
+
+            for(int noteCount = 0; noteCount < 4; noteCount++) // Converts incoming string arguments into Note form and checks for nulls
+            {
+                try
+                {
+                string _currentNoteString = args.RingtoneStringArgumentArray[noteCount].Replace("#", "sharp");
+                _RingtoneNoteArgumentArray[noteCount] = (Note)Enum.Parse(typeof(Note), _currentNoteString);
+                }
+                catch
+                {
+                    return;
+                }
+            }
+
+            UpdateRingerComponent(ringer, _RingtoneNoteArgumentArray);
+        }
+
+        public void RandomizeRingtone(EntityUid uid, RingerComponent ringer, ComponentInit args) //Randomizes the ringtone
+        {
+            Note[] _randomizedRingerNoteArr = new Note[4];
+            var randomNote = new Random();
+
+            for(int count = 0; count < 4; count++)
+            {
+                int randomNoteIndex = randomNote.Next(Enum.GetNames(typeof(Note)).Length - 1);
+                var currentNote = (Note)randomNoteIndex;
+                _randomizedRingerNoteArr[count] = currentNote;
+            }
+
+            UpdateRingerComponent(ringer, _randomizedRingerNoteArr);
+
+        }
+
+        //Non Event Functions
+
+        private bool UpdateRingerComponent(RingerComponent ringer, Note[] ringerNoteArr)
+        {
+            string[] _ringtonePathArr = new string[4];
+
+            for(int strngCount = 0; strngCount < 4; strngCount++)
+            {
+
+                if((int)ringerNoteArr[strngCount] < 12 )
+                {
+                    _ringtonePathArr[strngCount] = _pathIndex[ringerNoteArr[strngCount].ToString()];
+                }
+                else
+                {
+                    return false;
+                }
+            }
```

Comment:
Just store the ringtone as the notes on the ringer and not the string representation.

You'll also need to mark the enum as serializable + netserializable to be able to send it over the network.